transactions/redeemAllLostAndFoundNFTs.cdc

```diff
@@ -0,0 +1,49 @@
+import FindLostAndFoundWrapper from "../contracts/FindLostAndFoundWrapper.cdc"
+import LostAndFound from "../contracts/standard/LostAndFound.cdc"
+import FINDNFTCatalog from "../contracts/FINDNFTCatalog.cdc"
+import NFTCatalog from "../contracts/standard/NFTCatalog.cdc"
+import NonFungibleToken from "../contracts/standard/NonFungibleToken.cdc"
+import MetadataViews from "../contracts/standard/MetadataViews.cdc"
+import FIND from "../contracts/FIND.cdc"
+
+transaction() {
+
+	let receiverCaps : {String : Capability<&{NonFungibleToken.Receiver, MetadataViews.ResolverCollection}>} 
+	let ids : {String : [UInt64]}
+
+	prepare(account: AuthAccount){
+
+        let nftInfos : {String : NFTCatalog.NFTCollectionData} = {}
+		self.receiverCaps = {}
+
+		self.ids = FindLostAndFoundWrapper.getTicketIDs(user: account.address, specificType: Type<@NonFungibleToken.NFT>())
+
+		for type in self.ids.keys{ 
+
+            if nftInfos[type] == nil {
+                let collections = FINDNFTCatalog.getCollectionsForType(nftTypeIdentifier: type) ?? panic("NFT type is not supported at the moment. Type : ".concat(type))
+                nftInfos[type] = FINDNFTCatalog.getCatalogEntry(collectionIdentifier: collections.keys[0])!.collectionData
+            }
+            let nft = nftInfos[type]!
+
+			var targetCapability = self.receiverCaps[type]
+			if targetCapability == nil {
+				targetCapability = account.getCapability<&{NonFungibleToken.Receiver, MetadataViews.ResolverCollection}>(nft.publicPath)
+				if !targetCapability!.check(){
+					panic("Corresponding collection is not initialized. Type : ".concat(type))
```

Comment:
I think this code belongs in a contract if we want to make a generic one so that we can emit an event when this happen and not panic. If a user want to redeem everything they do not want it to fail if something has not been initialized.

Or have you talked with frontend team and they will check if things are not set up and do it accordingly?

Reply:
Currently, we have a script that returns all NFT types that are initiable (which are in NFTCatalog). 
Then we can borrow what's from Link to init the account storages before they redeem. 

But totally agree on emitting events than panicking here